docs/features/interceptors.md

```diff
@@ -0,0 +1,121 @@
+# Interceptors
+
+## Summary
+[summary]: #summary
+
+*Interceptors* are an experimental compiler feature. An *interceptor* is a method which can declaratively substitute a call to itself instead of a call to an *interceptable* method at compile time. This substitution occurs by having the interceptor declare the source locations of the calls that it intercepts. This provides a limited facility to change the semantics of existing code by adding new code to a compilation (e.g. in a source generator).
```

Comment:
>substitute a call to itself instead of a call to an *interceptable* method [](http://example.com/codeflow?start=106&length=74)

Perhaps "substitute a call to an *interceptable* method with a call to itself"